sdk/communication/AzureCommunicationCalling/Swift_Generated_Interfaces/AzureCommunicationCalling.h

```diff
@@ -0,0 +1,1124 @@
+    open var localVideoStreams: [LocalVideoStream]! { get }
+
+    
+    /// Indicates if recording is active in current call
+    open var isRecordingActive: Bool { get }
+
+    
+    /**
+     * The delegate that will handle events from the ACSCall.
+     */
+    unowned(unsafe) open var delegate: CallDelegate!
+
+    
+    /// Mute local microphone.
+    open func mute(completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Unmute local microphone.
+    open func unmute(completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Send DTMF tone
+    open func sendDtmf(tone: DtmfTone, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Start sharing video stream to the call
+    open func startVideo(stream: LocalVideoStream!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Stop sharing video stream to the call
+    open func stopVideo(stream: LocalVideoStream!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// HangUp a call
+    open func hangUp(options: HangUpOptions!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Remove a participant from a call
+    open func remove(participant: RemoteParticipant!, completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Hold this call
+    open func hold(completionHandler: ((Error?) -> Void)!)
+
+    
+    /// Resume this call
+    open func resume(completionHandler: ((Error?) -> Void)!)
+
+    
+    // Class extension begins for Call.
+    open var callerId: CommunicationIdentifier
+
+    open func add(participant: CommunicationIdentifier!) -> RemoteParticipant!
+
+    open func add(participant: PhoneNumberIdentifier!, options: AddPhoneNumberOptions!) -> RemoteParticipant!
+}
+
+// Class extension ends for Call.
+
+
+/// Describes a remote participant on a call
+open class RemoteParticipant : NSObject {
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    /// Private Preview Only: Display Name of the remote participant
+    open var displayName: String! { get }
+
+    
+    /// True if the remote participant is muted
+    open var isMuted: Bool { get }
+
+    
+    /// True if the remote participant is speaking
+    open var isSpeaking: Bool { get }
+
+    
+    /// Reason why participant left the call, contains code/subcode.
+    open var callEndReason: CallEndReason! { get }
+
+    
+    /// Current state of the remote participant
+    open var state: ParticipantState { get }
+
+    
+    /// Remote Video streams part of the current call
+    open var videoStreams: [RemoteVideoStream]! { get }
+
+    
+    /**
+     * The delegate that will handle events from the ACSRemoteParticipant.
+     */
+    unowned(unsafe) open var delegate: RemoteParticipantDelegate!
+
+    
+    // Class extension begins for RemoteParticipant.
+    open var identity: CommunicationIdentifier
+}
+
+// Class extension ends for RemoteParticipant.
+
+
+/// Describes the reason for a call to end
+open class CallEndReason : NSObject {
+
```

Comment:
Why is this not an enum?

Reply:
This was discussed during the review. There are too many codes and subcodes to create an effective enum. But, then how does the developer make sense of the code and subcode? Is there something that translates these codes into useful info?

Reply:
As discussed at the moment this is how it is for other platforms also its supposed to be. Closing this as of now as it wont be possible, when we fix other platforms in the future we will fix this one also.

Reply:
Add a string field here.